### DISCUSSION TITLE
Why does `reload-all` not refresh git gutters after using a Kitty overlay in a sequence of commands?

### DISCUSSION THREAD

**Discussion by David-Else** [helix-editor/helix#12077] [Q&A]
I have this shortcut to load lazygit in Kitty in an overlay, when I quit it returns to Helix:
```
[keys.normal.C-t]
l = [ ":sh kitty @ launch --no-response --cwd=current --type=overlay --copy-env lazygit", ":reload-all" ]
```
but the git gutters are not updated when I quit lazygit and return to Helix. Another user does something very similar and they are updated: https://github.com/helix-editor/helix/discussions/12045

I often find after reloading files after a git commit the git gutters don't update until I move the cursor, is this a know bug in Helix or some kind of side effect of using Lazygit? Are they related?

**Comment by gabydd** [comment]
I think this is cause :sh is async so the reload-all is ran right after the shell command is started you can use something like insert-output instead

**Reply by David-Else** [reply]
Thanks for the info! Do you have any idea why `insert-output` is not working in Kitty but OK in Westerm? I describe the problem here: https://github.com/helix-editor/helix/discussions/12045#discussioncomment-11277859

Kitty has printed Unicode input instead of performing key presses when I log onto remote shells on other machines before, I have confirmed that the terminal type is defiantly `xterm-kitty` inside the insert output buffer with `echo $TERM` inside the Helix list of commands.

**Comment by mil-ad** [comment]
I think you need `--wait-for-child-to-exit` flag in your `kitty @ launch` as well otherwise "reload-all" runs while lazygit is still open.
